### ISSUE TITLE
🧩 [Phase 2] Dummy Multimodal Dataset + DataLoader

### ISSUE THREAD

**Issue by shoaib-inamdar** [shoaib-inamdar/MultiModal-Cane-Disease-Detection#4]
**Parent:** #2 

## 🎯 Goal
Build a `SugarcaneDataset` class using **synthetic (fake) data** so the entire
pipeline can be developed and tested without real field images. When the real
Maharashtra dataset arrives, change one flag: `synthetic=False`.

## 🏗️ Sample Format
Each sample returns a dictionary:
```python
{
    "image":    torch.Tensor,  # shape [3, 224, 224], float32
    "metadata": torch.Tensor,  # shape [4], float32, values in [0, 1]
                               # [Temperature, Humidity, Soil Moisture, Rainfall]
    "label":    torch.Tensor,  # shape [], torch.long, value 0–3
                               # 0=Healthy, 1=Red Rot, 2=Grassy Shoot, 3=Smut
}
```

## 📋 Files to Create
- [ ] `configs/obj1_cross_attention.yaml` — all experiment settings (4 classes, Swin-Tiny, Colab paths)
- [ ] `src/utils/config.py` — `load_config(path: str) -> dict` using `yaml.safe_load`
- [ ] `src/utils/seed.py` — `set_seed(seed=42)` for reproducibility
- [ ] `src/data/augmentation.py` — `get_train_transforms()` and `get_val_transforms()`
- [ ] `src/data/dataset.py` — `SugarcaneDataset(torch.utils.data.Dataset)`
- [ ] `src/data/dataloader.py` — `get_dataloaders(config) -> dict`
- [ ] `tests/unit/test_dataset.py` — 4 smoke tests

## 🧪 Metadata Normalization (Synthetic)
```
Temperature  (15–45°C)   / 50  → [0, 1]
Humidity     (30–100%)   / 100 → [0, 1]
Soil Moisture(10–90%)    / 100 → [0, 1]
Rainfall     (0–200mm)   / 300 → [0, 1]
```

## 🗃️ Dataset Stages

| Stage | When | Action |
|-------|------|--------|
| **A - Now** | Phases 2–7 | Use this dummy/synthetic dataset |
| **B - Next** | After pipeline works | Set `synthetic=False`, use real Kaggle leaf images. Keep `synthetic_metadata=True` (no real weather yet) |
| **C - Final** | Objective 3 | Set both flags False, use real Maharashtra field data + DHT22 readings |

> Stage A dataset: [Sugarcane Leaf Disease (Maharashtra)](https://data.mendeley.com/datasets/9424skmnrk/1) — download when ready for Stage B.

## ✅ Done When
```
uv run pytest tests/unit/test_dataset.py -v
→ 4 passed ✅
```
Tests check: length, image shape `[3,224,224]`, metadata range `[0,1]`, batch shape.
